src/hotspot/share/runtime/os.cpp

```diff
@@ -1993,3 +1993,69 @@ void os::PageSizes::print_on(outputStream* st) const {
     st->print("empty");
   }
 }
+
+// Check minimum allowable stack sizes for thread creation and to initialize
+// the java system classes, including StackOverflowError - depends on page
+// size.
+// The space needed for frames during startup is platform dependent. It
+// depends on word size, platform calling conventions, C frame layout and
+// interpreter/C1/C2 design decisions. Therefore this is given in a
+// platform (os/cpu) dependent constant.
+// To this, space for guard mechanisms is added, which depends on the
+// page size which again depends on the concrete system the VM is running
+// on. Space for libc guard pages is not included in this size.
+jint os::set_minimum_stack_sizes(size_t os_min_stack_allowed) {
```

Comment:
Rather than pass in the minimum and spread that knowledge across each OS specific file, can we not just define a field in the os class and have it set appropriately (during os::init()?) - once from os_posix.cpp and once from os_windows.cpp?

Reply:
Could do that, sure.